File: app/src/main/java/org/dhis2/utils/UtilsModule.java
package org.dhis2.utils;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
@Singleton
public final class UtilsModule {

    @Provides
    @Singleton
    CurrentDateProvider currentDateProvider() {
        return new CurrentDateProviderImpl();
    }

    @Provides
    @Singleton
    RulesUtilsProvider rulesUtilsProvider() {
        return new RulesUtilsProviderImpl();
    }
}

File: app/src/main/java/org/dhis2/utils/UtilsModule.kt
package org.dhis2.utils

import javax.inject.Singleton

import dagger.Module
import dagger.Provides

@Module
@Singleton
class UtilsModule {

    @Provides
    @Singleton
    internal fun currentDateProvider(): CurrentDateProvider {
        return CurrentDateProviderImpl()
    }

    @Provides
    @Singleton
    internal fun rulesUtilsProvider(codeGenerator: CodeGenerator): RulesUtilsProvider {
        return RulesUtilsProviderImpl(codeGenerator)
    }
}
